app/scripts/modules/core/src/api/network.interceptor.ts

```diff
@@ -12,10 +13,12 @@ import {
 export class NetworkInterceptor implements IHttpInterceptor {
 
   private networkAvailable: IDeferred<void>;
-  private retryQueue: IRequestConfig[] = [];
+  private retryQueue: Dictionary<number> = {};
+  private MAX_RETRIES = 4;
 
   constructor(private $q: IQService,
               private $window: IWindowService,
+              private $timeout: ITimeoutService,
               private $injector: any) {
     'ngInject';
     this.$window.addEventListener('offline', this.handleOffline);
```

Comment:
I think you need `@BindAll()` on the class, or use `this.handleOffline.bind(this)`

Reply:
Only if we want it to work